src/ast_utils.h

```diff
@@ -46,6 +46,34 @@ struct BreakSeeker : public PostWalker<BreakSeeker, Visitor<BreakSeeker>> {
   }
 };
 
+// Finds all functions that are reachable via direct calls.
+
+struct DirectCallGraphAnalyzer : public PostWalker<DirectCallGraphAnalyzer, Visitor<DirectCallGraphAnalyzer>> {
+  Module *module;
+  std::vector<Function*> queue;
+  std::unordered_set<Function*> reachable;
+
+  DirectCallGraphAnalyzer(Module* module, std::vector<Function*> root) : module(module) {
+    for (auto curr : root) {
```

Comment:
Is this technically necessary? Or is it more of a style preference? I thought I only need to be careful when dealing with references.